diff --git a/cmake/cpp.cmake b/cmake/cpp.cmake
index 414d878242..dbef70ecad 100644
--- a/cmake/cpp.cmake
+++ b/cmake/cpp.cmake
@@ -27,6 +27,7 @@ if(USE_COINOR)
 endif()
 if(USE_GLPK)
   list(APPEND OR_TOOLS_COMPILE_DEFINITIONS "USE_GLPK")
+  set(GLPK_DIR glpk)
 endif()
 if(USE_PDLP)
   list(APPEND OR_TOOLS_COMPILE_DEFINITIONS "USE_PDLP")
@@ -226,6 +227,7 @@ foreach(SUBPROJECT IN ITEMS
  constraint_solver
  ${PDLP_DIR}
  ${GSCIP_DIR}
+ ${GLPK_DIR}
  glop
  graph
  gurobi
diff --git a/ortools/glpk/CMakeLists.txt b/ortools/glpk/CMakeLists.txt
new file mode 100644
index 0000000000..4643de4a0c
--- /dev/null
+++ b/ortools/glpk/CMakeLists.txt
@@ -0,0 +1,24 @@
+if(NOT USE_GLPK)
+  return()
+endif()
+
+file(GLOB _SRCS "*.h" "*.cc")
+
+set(NAME ${PROJECT_NAME}_glpk)
+
+# Will be merge in libortools.so
+#add_library(${NAME} STATIC ${_SRCS})
+add_library(${NAME} OBJECT ${_SRCS})
+set_target_properties(${NAME} PROPERTIES
+  CXX_STANDARD 17
+  CXX_STANDARD_REQUIRED ON
+  CXX_EXTENSIONS OFF
+  POSITION_INDEPENDENT_CODE ON
+  )
+target_include_directories(${NAME} PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}>)
+target_link_libraries(${NAME} PRIVATE
+  absl::strings
+  $<$<BOOL:${USE_GLPK}>:GLPK::GLPK>)
+#add_library(${PROJECT_NAME}::glpk ALIAS ${NAME})
